src/extension/ScriptRunner.ts

```diff
@@ -187,19 +187,126 @@ export class ScriptRunner {
     })
   }
 
-  private postProblem(str: string): void {
-    const dataList = str.split('\n')
-    const diagnosticDataList = dataList[2].split(':')
-    const rowPosition = parseInt(diagnosticDataList[1]) - 1
-    const colStartPosition = parseInt(diagnosticDataList[2]) - 1
-    const startPosition = new Position(rowPosition, colStartPosition)
-    const colEndPosition = colStartPosition + dataList[3].length - 1
-    const endPosition = new Position(rowPosition, colEndPosition)
-    const range = new Range(startPosition, endPosition)
-    const errorMsg = diagnosticDataList[4]
-    const diagnostic = new Diagnostic(range, errorMsg)
-    const diagnostics: Diagnostic[] = [diagnostic]
-    const uri = Uri.parse(diagnosticDataList[0])
-    this.diagnosticCollection.set(uri, diagnostics)
+  private postProblems(confFile: string): void {
+    const ignoreFolderRegex = '{certora-logs,conf}'
+    const resourceErrorsFile = 'resource_errors.json'
+    vscode.workspace
+      .findFiles(resourceErrorsFile, ignoreFolderRegex, 1)
+      .then(data => {
+        workspace.fs.readFile(data[0]).then(data => {
+          const decoder = new TextDecoder()
+          const content = decoder.decode(data)
+          const resource_error = this.getResourceError(content)
+
+          const pathRegex = /((\\|\/)[a-z0-9\s_@\-^!#$%&+=.{}]+)+/i
+          const locationRegex = /\d+:\d+/g
+
+          resource_error.topics.forEach(topic => {
+            if (topic.messages.length > 0) {
+              topic.messages.forEach(message => {
+                const curMessage: string = message.message
+
+                const path = pathRegex.exec(curMessage)
+                const location = locationRegex.exec(curMessage)
+                const problemPath = this.getPath(path, confFile)
+
+                const descriptiveMessage = (topic.name + ':' + curMessage)
+                  .replace(pathRegex, '')
+                  .replace(locationRegex, '')
+                  .replace(/:{2,}/g, ':')
+
+                const position: Position = this.getPosition(
+                  location,
+                  path,
+                  confFile,
+                )
+                const range: Range = new Range(position, position)
```

Comment:
the start position is equals to the end position because we are not getting the end position from the resource_errors.json file. I'm able to read the file where the problem originated from and get the length of the line if this is necessary (its not hard, but I'm not sure we want to waist resources on it... or maybe we do?)
vscode handles this range by emphasizing the word that starts at "position".
I added a short comment about this :)